pkg/operator/controller/externaldns/pod.go

```diff
@@ -269,6 +269,7 @@ func (b *externalDNSContainerBuilder) fillProviderSpecificFields(container *core
 
 // fillAWSFields fills the given container with the data specific to AWS provider
 func (b *externalDNSContainerBuilder) fillAWSFields(container *corev1.Container) {
+	container.Args = addTXTPrefixFlag(container.Args)
```

Comment:
time="2021-10-16T13:42:58Z" level=info msg="Desired change: CREATE hello-openshift.apps.misalunkexternaldns.feature CNAME [Id: /hostedzone/Z05215201EZME82PFS3MM]"
time="2021-10-16T13:42:58Z" level=info msg="Desired change: CREATE hello-openshift.apps.misalunkexternaldns.feature TXT [Id: /hostedzone/Z05215201EZME82PFS3MM]"
time="2021-10-16T13:42:58Z" level=error msg="Failure in zone misalunkexternaldns.feature. [Id: /hostedzone/Z05215201EZME82PFS3MM]"
time="2021-10-16T13:42:58Z" level=error msg="InvalidChangeBatch: [RRSet of type TXT with DNS name hello-openshift.apps.misalunkexternaldns.feature. is not permitted because a conflicting RRSet of type CNAME with the same DNS name already exists in zone misalunkexternaldns.feature.]\n\tstatus code: 400, request id: 5708442e-d0f8-4e0f-9fd5-92938db94245"
time="2021-10-16T13:42:58Z" level=error msg="failed to submit all changes for the following zones: [/hostedzone/Z05215201EZME82PFS3MM]"
time="2021-10-16T13:43:56Z" level=debug msg="Refreshing zones list cache"

Reply:
It's a bit unnecessary to add the TXT prefix for any source in AWS but this would be the simplest and harmless approach.